data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -121,7 +121,11 @@ public GenesisData getGenesisData() {
 
   public SafeFuture<Optional<ObjectAndMetaData<BlockHeader>>> getBlockHeader(
       final String slotParameter) {
-    return fromBlock(slotParameter, block -> new BlockHeader(block, true));
+    return fromBlock(
+        slotParameter,
+        block ->
+            new BlockHeader(
+                block, combinedChainDataClient.isCanonicalBlock(block.getSlot(), block.getRoot())));
```

Comment:
We should just get the canonical flag from the block meta data rather than recalculating it here (as we may have since re-orged and get inconsistent results).

Reply:
we've only got the block here, im not sure how you're getting to block metadata...

Reply:
The BlockSelector returns `BlockAndMetaData`.